### PR TITLE
Add mysqlclient

### PR DESCRIPTION
**Pull request by halldc**
Development on #5643 appears to have stalled, so I'm hoping we can get this merged instead. 

I've made the changes requested by @marcelotrevisani @jakirkham @synapticarbors and updated to the latest version.

@nehaljwani @pmlandwehr If you would like to be added as maintainers, please let me know.

### CODE REVIEW

File: recipes/mysqlclient/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+requirements:
+  build:
+    - python
+    - pip
+    - mysql-connector-c
+
+  run:
+    - python
+    - mysql-connector-c
```
Comment (nehaljwani) [inline recipes/mysqlclient/meta.yaml]: Doesn't mysql-connector-c have run_exports?

File: recipes/mysqlclient/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: ff8ee1be84215e6c30a746b728c41eb0701a46ca76e343af445b35ce6250644f
+
+build:
+  number: 0
+  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed --no-cache-dir -vvv"
+  skip: True  # [win]
+
+requirements:
+  build:
```
Comment (nehaljwani) [inline recipes/mysqlclient/meta.yaml]: Change this to 'host'

File: recipes/mysqlclient/meta.yaml (hunk 1 above)
Comment (halldc) [reply recipes/mysqlclient/meta.yaml]: Doesn't look like it (see [recipe](https://github.com/conda-forge/mysql-connector-c-feedstock/blob/master/recipe/meta.yaml)).

**Comment by bgruening** [comment]
Thanks @halldc!

**Pull request merged**